c10/xpu/XPUStream.cpp

```diff
@@ -37,27 +37,30 @@ thread_local std::unique_ptr<StreamId[]> current_streams = nullptr;
 // ~~~~~~~~~~~~~~~~~~~~~~~~~~
 // How do we assign stream IDs?
 //
-// -- 56 bits --  -- 5 bits -----  -- 3 bits --  -- 1 bits --
-//     zeros      StreamIdIndex    StreamIdType  Ext/native stream
+// -- 55 bits --    -- 5 bits --     -- 3 bits --     -- 1 bit --
+//     zeros       StreamIdIndex     StreamIdType    Ext/native stream
+//                ignored for ext   ignored for ext
 //
 // Where StreamIdType:
 //  000 = normal priority queue
 //  001 = high priority queue
+//  111 = external queue
 //
-// for external stream, StreamID is a sycl::queue* pointer
-// this means that last bit will always be 0
-// so when constructing StreamId for a native stream we set last bit to 1
-// to distinguish between native and external streams
+// For external stream, StreamID is a sycl::queue* pointer. This means that last
+// bit will always be 0. So when constructing StreamId for a native stream we
```

Comment:
Why is the last bit always 0 for an external stream? Is there a spec to guarantee the fact? I know it is almost true for most cases. However, we'd better check with the Intel SYCL compiler community to ensure that the pointer of the SYCL queue is always 64byte/32byte aligned.

Reply:
This is irrelated to the SYCL compiler. The pointer on how to align is specified to the platform. For instance GCC, it always aligned to a multiple of eight bytes, see https://ftp.gnu.org/old-gnu/Manuals/glibc-2.2.3/html_node/libc_30.html#:~:text=The%20address%20of%20a%20block,functions%20are%20declared%20in%20%60stdlib.. 
I add a `TORCH_CHECK` on https://github.com/pytorch/pytorch/pull/142347/commits/7b90bef59f734b1e0103a2b6503fcb90f84efba1 to ensure this design. And we can refine this design if a counter case happens.

Reply:
Do you mean the pointer does not come from the device runtime?